### PR TITLE
Storage options copy

### PR DESCRIPTION
**Pull request by will-moore**
This replaces #197, fixing merge conflicts between that and master and adding tests (described on the PR).

It also fixes an invalid comment and fixes a warning:

```
/Users/wmoore/opt/anaconda3/envs/omeroweb/lib/python3.9/site-packages/zarr/creation.py:221: UserWarning: ignoring keyword argument 'array_key'
  warn('ignoring keyword argument %r' % k)
```

### CODE REVIEW

**Review by sbesson — COMMENTED** [review]
One API question about the `da.to_zarr` workflow. Otherwise, definitely valuable to have this fix tested and merged in preparation of a final release with Dask support.

I think the amended test covers the basic scenario passing a chunk size as a single `storage_options` dictionary. I suspect next action is to wait until we hear about https://github.com/ome/ome-zarr-py/issues/219#issuecomment-1236368638 before merging?

Finally, re-reading https://github.com/ome/ome-zarr-py/pull/197#pullrequestreview-970028104, I think we are still lacking the coverage of `storage_options` passed as a list of dictionaries. If it cannot easily be added in this PR, I'd propose to at least capture this as an issue so that we can come back to it.

File: ome_zarr/writer.py [hunk 1]
```diff
@@ -246,7 +246,6 @@ def write_multiscale(
                 data = da.array(data).rechunk(chunks=chunks_opt)
                 options["chunks"] = chunks_opt
             da.to_zarr(
-                array_key=path,
```
[the remark this answers is not in the document]
Comment (will-moore) [reply ome_zarr/writer.py:249]: It fixes a warning:

```
/Users/wmoore/opt/anaconda3/envs/omeroweb/lib/python3.9/site-packages/zarr/creation.py:221: UserWarning: ignoring keyword argument 'array_key'
  warn('ignoring keyword argument %r' % k)
```
`array_key` is not used by `to_zarr()`: https://docs.dask.org/en/stable/generated/dask.array.to_zarr.html

**Comment by will-moore** [comment]
@sbesson Added test for `storage_options` as a list

**Comment by will-moore** [comment]
This PR is needed to fix the tests recently added to https://github.com/ome/ome-zarr-py/pull/220

**Review by sbesson — APPROVED** [review]

**Pull request merged**